src/main/scala/dpla/ingestion3/mappers/providers/PaMapping.scala

```diff
@@ -78,12 +81,12 @@ class PaMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeSeq
 
   override def dataProvider(data: Document[NodeSeq])
                            (implicit msgCollector: MessageCollector[IngestMessage]): EdmAgent = {
-    val contributors = extractStrings(data \ "metadata" \\ "contributor")
-    if (contributors.nonEmpty)
-      nameOnlyAgent(contributors.last)
-    else
-      throw new Exception(s"Missing required property dataProvider because " +
-        s"dc:contributor is empty for ${getProviderId(data)}")
+    extractStrings(data \ "metadata" \\ "contributor").lastOption match {
+      case Some(lastContributor) => nameOnlyAgent(lastContributor)
+      case None =>
+        msgCollector.add(missingRequiredError(getProviderId(data), "dataProvider"))
+        nameOnlyAgent("") // FIXME this shouldn't have to return an empty value.
```

Comment:
Just flagging this `FIXME` comment in case you meant to fix it sooner rather than later.